include/h2o/absprio.h

```diff
@@ -22,18 +22,17 @@
 #ifndef h2o__absprio_h
 #define h2o__absprio_h
 
-typedef enum en_h2o_absprio_urgency_type_t {
-    H2O_ABSPRIO_URGENCY_PREREQUISITE = 0,
-    H2O_ABSPRIO_URGENCY_DEFAULT = 1,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_0 = 2,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_1 = 3,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_2 = 4,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_3 = 5,
-    H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_4 = 6,
-    H2O_ABSPRIO_URGENCY_BACKGROUND = 7,
-} h2o_absprio_urgency_type_t;
+#define H2O_ABSPRIO_URGENCY_PREREQUISITE 0
+#define H2O_ABSPRIO_URGENCY_DEFAULT 1
+#define H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_0 2
+#define H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_1 3
+#define H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_2 4
+#define H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_3 5
+#define H2O_ABSPRIO_URGENCY_SUPPLEMENTARY_4 6
+#define H2O_ABSPRIO_URGENCY_BACKGROUND 7
+#define H2O_ABSPRIO_URGENCY_MAX H2O_ABSPRIO_URGENCY_BACKGROUND
 
-void h2o_absprio_parse_priority(const h2o_iovec_t *header_value, uint8_t *urgency, int *incremental);
+void h2o_absprio_parse_priority(const char *s, size_t len, uint8_t *urgency, int *incremental);
```

Comment:
It is our convention to flatten `const char *` and `size_t` when passing a vector, because doing so avoids the caller building a temporary structure.